fanaSystem/fanaAuthenticator/views.py

```diff
@@ -1,24 +1,62 @@
+from django.core.cache import cache
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from rest_framework import status
-from rest_framework_simplejwt.tokens import RefreshToken
+from .utils import send_otp , generate_otp
+from fanaSystem.utils import generic_error_handler
+from django.http import JsonResponse
+import jwt
+import datetime
+from decouple import config
 
-class CustomTokenObtainPairView(APIView):
-    def post(self, request, *args, **kwargs):
-        username = request.data.get("username")
-        password = request.data.get("password")
-        app = request.data.get("app")  # Get the app identifier from the request
 
-        # Validate credentials (replace with actual validation logic as needed)
-        if validate_credentials(username, password):
-            print("Credentials validated successfully.")
 
-            # Generate JWT token with an additional claim for the app
-            return Response(generate_tokens(username=username, app=app))
 
+class CustomTokenObtainPairView(APIView):
+    def post(self, request, *args, **kwargs):
+        phone_number = request.data.get("phone")
+        otp = generate_otp(6)
+        response = send_otp(phone_number, otp)
+        cache.add(phone_number , otp , 3600 )
+        if response["success"]:
+            return Response({"success" : True , "message" : response["message"]})
         else:
-            return Response({"status": "error", "detail": "Invalid credentials"}, status=status.HTTP_401_UNAUTHORIZED)
-
+            return Response({"success" : False , "error-source" : response["error"] , "message" : response["message"]})
+            
+class VerifyOtp(APIView):
+    def post(self, request, *args, **kwargs):
+        try:
+            user_otp = request.data.get("otp")
+            phone_number = request.data.get("phone")
+            actual_otp = cache.get(phone_number)
+            # actual_otp = user_otp
+            print(actual_otp , user_otp)
+            if actual_otp == user_otp and user_otp != None:
+                payload = {
+                    'phone_number': phone_number,
+                    'exp': datetime.datetime.now() + datetime.timedelta(hours=1)  # Set expiration to 1 hour
+                }
+                access_token = jwt.encode(payload,config('S_KEY'), algorithm='HS256')
+                response = JsonResponse({"success" : True , "message" : "OTP Verified"} , status=200)
```

Comment:
Is this working with the fanaDashboard (the dashboard for manager)
Can you make changes in the login.html too.

Reply:
Yes , I have checked , its working fine.